runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -329,16 +331,8 @@ public void clear(K key, W window) {
           }
         }
 
-        private void prepareStateBackend(K key, Coder<K> keyCoder) {
-          final ByteBuffer encodedKey;
-          try {
-            // We need to have NESTED context here with the ByteStringCoder.
-            // See StateRequestHandlers.
-            encodedKey =
-                ByteBuffer.wrap(CoderUtils.encodeToByteArray(keyCoder, key, Coder.Context.NESTED));
-          } catch (CoderException e) {
-            throw new RuntimeException("Couldn't set key for state");
-          }
+        private void prepareStateBackend(K key) {
+          final ByteBuffer encodedKey = ByteBuffer.wrap(((ByteString) key).toByteArray());
```

Comment:
@mxm there is a follow-up here to express ByteString through the type variables. Couldn't figure out how to do that yet, maybe requires change in `StateRequestHandlers`?

Reply:
I got that to work here: https://github.com/tweise/beam/commit/f988e043b4c2b999974a18a9f9d4976a6287973d

But it may require changes in other runners also and does not fit into this PR.